kubetest2-gke/deployer/common.go

```diff
@@ -74,3 +95,22 @@ func (d *deployer) initialize() error {
 
 	return nil
 }
+
+func filterMultiProjectClusterName(clusters []string, projectID int) ([]string, error) {
+	result := make([]string, 0)
+	for _, clusterName := range clusters {
+		parts := strings.Split(clusterName, "-")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("cluster name does not follow expected format (name-projectIndex): %s", clusterName)
+		}
+		projectIndex, err := strconv.Atoi(parts[1])
+		if err != nil {
+			return nil, fmt.Errorf("cluster name does not follow contain a valid project index (name-projectIndex. E.g: cluster-0): %s", err)
+		}
+		if projectIndex == projectID {
+			// Network field is empty
+			result = append(result, parts[0])
+		}
```

Comment:
insteading of erroring if the syntax is not provided, a fallback where a random project is chosen from the provided list sounds more flexible?

Reply:
Hmm I'm not sure if that flexibility is desirable, if users decide to use multi-project profile, it should be required to specify which project to use to create which clusters, otherwise the result won't be predictable, e.g. it could be possible to have all clusters randomly created in one project and run into quota issues. So I left this unchanged.

Reply:
makes sense let's lave that for now. I was mostly worried about finding out too late that the user supplied syntax is wrong, but as long as we validate it early enough should be fine.